### PR TITLE
Export isSonarQubeAvailable

### PR DESCRIPTION
**Pull request by bobalong79**
Signed-off-by: Rob Long <Robert.Long@lv.com>

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

Allows users to optionally include the `SonarQubeCard` in their `EntityPage.tsx`, e.g.

```ts
import {
  SonarQubeCard,
  isSonarQubeAvailable
} from '@backstage/plugin-sonarqube';

...

const overviewContent = (
  <Grid container spacing={3} alignItems="stretch">
    <Grid item xs={8}>
      <EntityAboutCard variant="gridItem" />
    </Grid>
    <EntitySwitch>
      <EntitySwitch.Case if={isSonarQubeAvailable}>
        <Grid item xs={4}>
          <SonarQubeCard />
        </Grid>
      </EntitySwitch.Case>
    </EntitySwitch>
);
```

### CODE REVIEW

**Review by adamdmharvey — COMMENTED** [review]
Nice one! 👍🏻 

Before exporting, wonder if we should align the plugin to the new compatibility system which has a naming convention for those cards?  That way it's easier to refactor now that it's bounded by the plugin, before exporting it and causing breaking changes in future.

https://backstage.io/docs/plugins/composability#naming-patterns

Willing to tackle that in this PR?

**Comment by bobalong79** [comment]
Happy to give it a go! Can you point me to a plugin that I can use for a reference?

**Comment by adamdmharvey** [comment]
Great!  Yeah not super complex I don't think, but if you search for that named card you gave an example to, it should be named everywhere with `Entity*Card`.  So Jenkins has a similar thing, where there's both a content page AND a card, and like you suggested here, you can elevate just the Jenkins Latest card to the main entity page.  So it's plugin looks like this:

https://github.com/backstage/backstage/blob/8a121a5e2ff2b32c1c9527a754b7e1a5796d7e05/plugins/jenkins/src/index.ts#L17-L22

So just following that pattern. Depending on your editor/IDE you might even have a one click "refactor" command that would find all references to it and refactor them. Then you can run local tests to ensure everything lines up, and see what changes it causes?

File: .changeset/slimy-llamas-search.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@lvfs/plugin-sonarqube': patch
```
Comment (benjdlambert) [inline .changeset/slimy-llamas-search.md:2]: Think that this should be `'@backstage/plugin-sonarqube': patch`

Comment (bobalong79) [reply .changeset/slimy-llamas-search.md:2]: Whoops, will change this, thanks!

**Comment by Fox32** [comment]
hmm, was this already done in #5245?

**Comment by Fox32** [comment]
> Great! Yeah not super complex I don't think, but if you search for that named card you gave an example to, it should be named everywhere with `Entity*Card`. So Jenkins has a similar thing, where there's both a content page AND a card, and like you suggested here, you can elevate just the Jenkins Latest card to the main entity page. So the Jenkins plugin looks like this:
> 
> https://github.com/backstage/backstage/blob/8a121a5e2ff2b32c1c9527a754b7e1a5796d7e05/plugins/jenkins/src/index.ts#L17-L22
> 
> So just following that pattern. Depending on your editor/IDE you might even have a one click "refactor" command that would find all references to SonarQubeCard and refactor them as EntitySonarQubeCard? (I think; and the core maintainers can triple check me on that.) Then you can run local tests to ensure everything lines up, and see what changes it causes?

The plugin is already ported to the new model and correctly exports `EntitySonarQubeCard`: https://github.com/backstage/backstage/blob/master/plugins/sonarqube/src/plugin.ts#L41-L48

**Comment by bobalong79** [comment]
@Fox32 Yes you're right, sorry for the confusion, I was working with the currently release package not master :)

**Pull request closed without merging**